packages/astro/src/core/create-vite.ts

```diff
@@ -24,7 +24,7 @@ export type ViteConfigWithSSR = vite.InlineConfig & { ssr?: vite.SSROptions };
 interface CreateViteOptions {
 	astroConfig: AstroConfig;
 	logging: LogOptions;
-	mode: 'dev' | 'build';
+	mode: 'dev' | 'build' | string;
```

Comment:
when will this be something else?

Reply:
Vitest automatically sets Vite's `mode` to `"test"`.

Reply:
Can we make this `| 'test'` then? Would be nice to still get error messages if you pass an invalid mode.